File: Apuntes/Nodos/Publicador.py
"""
TODO: Creacion de un node el que enviara informacion a un topic
"""

#!/usr/bin/env python
import rospy
from std_msgs.msg import Int32

rospy.init_node("Primer nodo") #Para crear un nodo

pub = rospy.Publisher("laser", Int32, queue_size=10) #Para crear un topic y si existe el topic solo se conecta a el
a = Int32() #Comfiguramos a como un mensaje de tipo Int32
contador = 0 #Creamos una variable contador
rate = rospy.Rate(10) #Para crear un rate de 10Hz

while not rospy.is_shutdown(): #Mientras no se cierre el nodo 
    contador += 1
    a.data = contador
    pub.publish(a) #Publicamos el mensaje
    rate.sleep() #Esperamos el tiempo necesario para que el rate sea de 10Hz




rospy.spin() #Para que el nodo no se cierre, bucle infinito

File: Apuntes/C++/Topics/publicador.cpp
#include <ros/ros.h>
#include <std_msgs/Int32.h>
#include <iostream>
using namespace std;

int main(int argc, char ** argv){

    //Creamos el nodo principal del programa
    ros::init(argc, argv, "Nodo publicador");
    ros::NodeHandle nh;

    //Creamos el Publicador
    ros::Publisher pub = nh.advertise<std_msgs::Int32>(contador, 1);

    //Creacion del mensaje
    std_msgs::Int32 cont;
    cont.data = 0;

    //Creacion de la frecuencia de publicacion 
    ros::Rate loop_rate(1);

    //Bucle de publicacion
    while (ros::ok()){
        pub.publish(cont);
        cont.data++;
        loop_rate.sleep();
    }

    return 0;
}